### PR TITLE
Add check for revoked iterations to HCP Packer datasources

### PR DESCRIPTION
**Pull request by sylviamoss**
### :hammer_and_wrench: Description

This adds a check for revoked iterations. For HCP Packer GA, revoking iterations will be possible and if an iteration is revoked it must never be used and the `terraform plan|apply` should fail. 

Since I updated the hcp-sdk-go version, I had to make some changes to vault's import and tests. Changes were checked and validated with @kkredit 

### :ship: Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-hcp/blob/main/CHANGELOG.md):

```release-note
* datasource/hcp_packer_image: Add check for revoked iterations [GH-240]
* datasource/hcp_packer_iteration: Add check for revoked iterations [GH-240]
* datasource/hcp_packer_image_iteration: Add check for revoked iterations [GH-240]
```

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAcc_dataSourcePacker'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/... -v -run=TestAcc_dataSourcePackerImage -timeout 120m
?       github.com/hashicorp/terraform-provider-hcp/internal/clients    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/consul     (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/input      (cached) [no tests to run]
=== RUN   TestAcc_dataSourcePackerImage
--- PASS: TestAcc_dataSourcePackerImage (27.25s)
=== RUN   TestAcc_dataSourcePackerImage_revokedIteration
--- PASS: TestAcc_dataSourcePackerImage_revokedIteration (13.53s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   41.102s
...
```

### CODE REVIEW

**Comment by kkredit** [comment]
The Vault-related changes look great. Thanks Moss!

**Review by bcmdarroch — COMMENTED** [review]
Had a couple questions!

**Review by nywilken — APPROVED** [review]
This looks good to me. Nicely done; especially with the added tests.

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320
 	github.com/hashicorp/go-version v1.3.0
 	github.com/hashicorp/hcl/v2 v2.8.2 // indirect
-	github.com/hashicorp/hcp-sdk-go v0.14.0
+	github.com/hashicorp/hcp-sdk-go v0.15.1-0.20220112153249-f565607d7cc4
```
[the remark this answers is not in the document]
Comment (sylviamoss) [reply go.mod:15]: Got it! So I'll wait for the release to update this one with 0.16.0

File: internal/provider/data_source_packer_iteration.go [hunk 2]
```diff
@@ -98,13 +99,18 @@ func dataSourcePackerIterationRead(ctx context.Context, d *schema.ResourceData,
...
+	if channel.Iteration == nil {
 		return diag.Errorf("no iteration information found for the specified channel %s", channelSlug)
 	}
 
-	iteration := channel.Pointer.Iteration
+	iteration := channel.Iteration
+	if !time.Time(iteration.RevokeAt).IsZero() {
+		// If RevokeAt is not a zero date, it means this iteration is revoked and should not be used
+		// to build new images.
+		return diag.Errorf("the iteration %s is revoked and can not be used", iteration.ID)
```
[the remark this answers is not in the document]
Comment (sylviamoss) [reply internal/provider/data_source_packer_iteration.go:110]: I think we can, I will try to think of a better text!

[the remark this answers is not in the document]
Comment (sylviamoss) [reply internal/provider/data_source_packer_iteration.go]: Done! I also improved some of the tests

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcmdarroch) [reply go.mod:15]: https://github.com/hashicorp/hcp-sdk-go/releases/tag/v0.16.0 ✨

**Review by bcmdarroch — APPROVED** [review]
Once you get the Go SDK v0.16.0 in, this is good to go!

**Pull request merged**